indexer/services/ender/src/validators/stateful-order-validator.ts

```diff
@@ -237,9 +239,37 @@ export class StatefulOrderValidator extends Validator<StatefulOrderEventV1> {
   }
 
   /**
-   * Skip order uuids in config env var.
+   * Skip SQL processing for
+   * - order uuids to be skipped in config env var.
+   * - vault stateful orders.
    */
-  public shouldExcludeEvent(): boolean {
+  public shouldSkipSql(): boolean {
+    const orderUUIDsToSkip: string[] = config.SKIP_STATEFUL_ORDER_UUIDS.split(',');
+    if (orderUUIDsToSkip.length === 0) {
+      return false;
```

Comment:
Does this prematurely return `false` regardless of if order is from `vault`, as long as `SKIP_STATEFUL_ORDER_UUIDS` is non-empty?
Additionally, we should reuse code related to `SKIP_STATEFUL_ORDER_UUIDS` between `shouldSkipSql` and `shouldSkipHandlers`